getodk/collect
# Replace Espresso tests for default image with Widget tests

Follow up to #3579

This adds tests for widgets that do and don't support default images. These tests are at the individual widget level and allow us to remove the Espresso test that was covering this feature previously. We'd initially discussed also adding tests for ther non default image flow but I think this is enough to merge for now (I've create an issue for the extra work at #3643).

#### What has been done to verify that this works as intended?

To do something like this generally the process I follow is:
1. Run the Espresso test I think we can get rid of - verify it's green
2. Comment out code for feature until the Espresso test is red
3. Write tests that drive out the feature at a different level - verify they're red
4. Make the new tests go green
5. Verify the Espresso test is also now green - delete it

#### Why is this the best possible solution? Were any other approaches considered?

We could of course keep the tests at an Espresso level instead but I feel like a widget/Robolectric test is better as it allows more granularity around the assertions (the image loaded for instance) and also provides faster and more "connected" feedback - it's easier to know what tests to run when working on a specific widget.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

No changes to non-test code so probably good to merge this on acceptance.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)
